src/components/ContactAdd.tsx

```diff
@@ -0,0 +1,40 @@
+import { FormEvent } from 'react';
+import { v4 as uuidv4 } from "uuid";
+import { Contact } from "../types";
+
+interface ContactAddProps {
+  onAddNew: (newContact: Contact) => void;
+}
+
+export default function ContactAdd({ onAddNew }: ContactAddProps) {
+
+  const handleAddNewContact = (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    const formData = new FormData(e.currentTarget);
+    const newContact = {
+      id: uuidv4(),
+      name: formData.get('name') as string,
+      city: formData.get('city') as string,
+    };
+
+    onAddNew(newContact);
+
+    e.target.reset();
```

Comment:
I noticed this line is highlighted with red in VSCode due to Typescript not recognising that `e.target` has a `reset` method. You can fix that by telling TS this is a form element: `(e.target as HTMLFormElement).reset()`